pulp_rpm/app/metadata_parsing.py

```diff
@@ -125,6 +126,8 @@ def process_other_package_element(element):
             text = subelement.text
             changelogs.append((author, date, text))
 
+    # changelogs listed in chronological order, grab the last N changelogs from the list
+    changelogs = changelogs[-settings.KEEP_CHANGELOG_LIMIT :]
```

Comment:
Did this change affect previous memory improvements (when you didn't process all entries, less were kept in memory I guess)?

Reply:
I haven't re-tested but I doubt it.  At most it would be by one package worth of changelogs.